3rdparty/everest/CMakeLists.txt

```diff
@@ -1,16 +1,16 @@
-list (APPEND everest_src)
-list (APPEND everest_inc_public)
-list (APPEND everest_inc)
-list (APPEND everest_def)
+add_library(everest
```

Comment:
Same comment for this file as for 3rdparry/CMakeLists.txt. This is not a real project and I suggest turning it to a `.cmake` file.

Reply:
As above, but this would have the downside of putting compiled binaries into the incorrect build subdirectory for no real gain.

Reply:
" incorrect build subdirectory "
Where does the cmake documentation forbids calling add_library() from an included file? If this is a supported feature how and why is the binary directory where the file will be build incorrect?

Reply:
It's not forbidden, but the most common use of `add_library` inside included scripts that I am aware of is to generate parametrised targets. In this case though there's only one `everest` library and it has a dedicated subdirectory, so add_subdirectory` seems like the perfect fit.

RE: the binary directory, `add_subdirectory` mirrors the build system hierarchy so that target locations are predictable. With `add_subdirectory`, we know that `libeverest.a` can be found in `${PROJECT_BINARY_DIR}/3rdparty/everest`, next where all of its intermediate files were generated (`${PROJECT_BINARY_DIR}/3rdparty/everest/CMakeFiles`).

Anyway, as above - discussion on this ought to happen on the dedicated thread rather than this unrelated change.